### PR TITLE
Internalize pyaction

### PR DESCRIPTION
**Pull request by joakim-hove**
With this PR the Python code of the PYACTION is loaded from an external file; i.e. the PYACTIONkeyword should be like this:

```
PYACTION
   ACTION_NAME  SINGLE  /  -- Metadata about the action
   path/to/python/action.py'   /  -- Filename of python code
```

mixing .DATA style metadata and Python code in the same keyword was just to ugly for the parsing process.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/Schedule/Action/PyAction.cpp [hunk 1]
```diff
@@ -28,12 +29,45 @@ using dict = int;
...
+
+    throw std::invalid_argument("RunCount string: " + run_count + " not recognized ");
+}
+
+
+std::string PyAction::load(const std::string& input_path, const std::string& fname) {
+    namespace fs = Opm::filesystem;
+    fs::path code_path = fs::path(input_path) / fs::path(fname);
+    if (fs::exists(code_path)) {
+        std::ifstream ifs(code_path.c_str(), std::ios::in);
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/Schedule/Action/PyAction.cpp]: You're supposed to be able to construct a `basic_ifstream` object directly from an `fs::path` object.  Did the following not work for you?
```C++
std::ifstream ifs(code_path);
```

File: tests/EMBEDDED_PYTHON.DATA [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+   300*1 /
+
+PERMZ
+   300*1 /
+
+SCHEDULE
+
+PYACTION
+  WCLOSE UNLIMITED /
+  'wclose.py' /
```
Comment (bska) [inline tests/EMBEDDED_PYTHON.DATA]: Do we no longer need `PYEND`?

Comment (joakim-hove) [reply tests/EMBEDDED_PYTHON.DATA]: This is described in some detail in the #1591 PR description; point is that the Python code is relegated to a separate file and then we no longer need `PYEND`; it just became too complex to handle both the metadata part:
```
WCLOSE UNLIMITED /
```
and the literal Python code in the same keyword in the `.DATA` file. I don't know if it is an improvement or - I guess it depends on the eyes of the beholder.

File: src/opm/parser/eclipse/EclipseState/Schedule/Action/PyAction.cpp (hunk 1 above)
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/Action/PyAction.cpp]: I did not know it was possible - I will update.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by bska** [comment]
Do you want to merge this instead of or in addition to #1591?

**Comment by joakim-hove** [comment]
I want to merge this - but I lost steam before starting on the soon-to-be-removed serialization.

**Comment by bska** [comment]
> I want to merge this

Okay. In that case I take it we'll close #1591 at the same time?

> I lost steam before starting on the soon-to-be-removed serialization.

Does that mean that there is more work needed here, or can it be merged in its current state?

**Comment by joakim-hove** [comment]
> Okay. In that case I take it we'll close #1591 at the same time?

Yes - I can close #1591 right away.

> Does that mean that there is more work needed here, or can it be merged in its current state?

This can be merged - but the serialization stuff is missing; so all in all it can not be merged...

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
@akva2 : I have made my first new-style serializer updates, can we on a general basis assume that an Integration test somewhere will barf at is if we get this wrong?

**Comment by akva2** [comment]
well. that's the last remaining catch. we still have to write those tests, and those tests sit in opm-simulators.

to bring this close to the code as well, we could have a static member to create an object to try for serialization. while also ugly, this would directly replace the constructors that we still have to keep updated in order to create these objects.

**Comment by joakim-hove** [comment]
> we could have a static member to create an object to try for serialization. while also ugly

Thank you for the explanation. I like that approach.

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=2489 please

**Pull request merged**